### DISCUSSION TITLE
Chain type should only have the "id" as mandatory

### DISCUSSION THREAD

**Discussion by wighawag** [wevm/viem#671] [General]
Currently the chain type include many field including names, rpcUrls, etc... as mandatory

the chainId should be the only thing mandatory since some provider do not need the rest, and actually I can cast `{id: myChainId} as Chain` succesfuly in the viem api that expect a chain object that  I tested (except of course the one to switch network and add network)

Is there a rationale why the Chain type include this or can we hope this to be simplified ?

**Comment by jxom** [comment]
We can probably simplify it. The `Chain` type was originated from wagmi, and wagmi needs RPC URLs, etc... However, you're right in saying that the only attribute viem really needs is the Chain ID. Can look into it!

**Comment by wighawag** [comment]
Any update on this ?@jxom ?

**Comment by glitch-txs** [comment]
IMO chain registry should follow EIP-3085 or close. Ideally this should be sent by the wallet but welp.
https://eips.ethereum.org/EIPS/eip-3085
